src/graph/graphcontext.c

```diff
@@ -255,15 +255,35 @@ int GraphContext_AddIndex(Index **idx, GraphContext *gc, const char *label, cons
 	// Retrieve the schema for this label
 	Schema *s = GraphContext_GetSchema(gc, label, SCHEMA_NODE);
 	if(s == NULL) s = GraphContext_AddSchema(gc, label, SCHEMA_NODE);
-	return Schema_AddIndex(idx, s, field, type);
+	int res = Schema_AddIndex(idx, s, field, type);
+	ResultSet *result_set = QueryCtx_GetResultSet();
+	if(res == INDEX_OK) {
+		if(result_set->stats.indicies_created == NOT_SET) {
+			result_set->stats.indicies_created = 1;
+		} else result_set->stats.indicies_created += 1;
```

Comment:
Are there queries that can generate more than 1 index? In any case, stylistically I think that an if/else block should either use curly braces or not; mixing them seems odd.

Reply:
fixed